x/observer/keeper/chain_nonces.go

```diff
@@ -14,14 +16,14 @@ import (
 func (k Keeper) SetChainNonces(ctx sdk.Context, chainNonces types.ChainNonces) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ChainNoncesKey))
 	b := k.cdc.MustMarshal(&chainNonces)
-	store.Set(types.KeyPrefix(chainNonces.Index), b)
+	store.Set(types.KeyPrefix(strconv.FormatInt(chainNonces.ChainId, 10)), b)
```

Comment:
Nit: we can add a function to create the Key from the chain id

Reply:
f8734e25e